proposals/Imperative-Shadow-DOM-Distribution-API.md

```diff
@@ -93,35 +134,33 @@ Other steps don't need to be updated from the standard's perspective, I think, t
 
 4. If the open flag is set and shadow’s mode is not "open", then return null.
 
-5. [New Step] If shadow's *slotting* is *manual*, return the first slot in shadow’s tree whose *manually-assigned-nodes* includes slotable, if any, and null otherwise.
+5. [New Step] If shadow's `slotAssignment` is "manual", return the associated slot in shadow’s tree whose `manually-assigned-nodes` includes slotable, if any, and null otherwise.
 
-6. Otherwise, return the first slot in shadow’s tree whose name is slotable’s name, if any, and null otherwise. (<= No change)
+6. Return the first slot in shadow’s tree whose name is slotable’s name, if any, and null otherwise. (<= No change)
 
 Note: This change implies:
 
--   *manually-assigned-nodes* should be considered an implementation detail. As long as the external behavior doesn't change, UA doesn't allocate *manually-assigned-nodes* for a slot.
--   *manually-assigned-nodes* is used only when a slot is in a shadow tree whose *slotting* is *manual*.
--   *manually-assigned-nodes* is not used when a slot is in a shadow tree whose *slotting* is *auto*.
+-   `manually-assigned-nodes` should be considered an implementation detail. As long as the external behavior doesn't change, UA doesn't allocate `manually-assigned-nodes` for a slot.
+-   `manually-assigned-nodes` is used only when a slot is in a shadow tree whose `slotAssignment` is "manual".
+-   `manually-assigned-nodes` is not used when a slot is in a shadow tree whose `slotAssignment` is "auto".
 
-    Web developers can call `slot.assign(...)` for such a slot, however, it is a sort of *no-op*, at least until the slot is moved to another shadow tree with 'manual', where *manually-assigned-nodes* might have a meaning (but it is unlikely).
-
--   If the same node is set to *manually-assigned-nodes* of more than one slots, the first slot in tree-order takes that node. The slot's location in the tree matters, as declarative API does so.
+     Web developers can not call `slot.assign(...)` for such a slot, it will throw exception to keep it consistent with when invalid nodes are passed in the "manual" case.
 
+-   If the same node is set to `manually-assigned-nodes` when it's already an assigned node in another slot, the node will removed from the previous slot and assigned to the new slot. The "slotchange" event will be raised for both slots.
 
 [To find a slot]: https://dom.spec.whatwg.org/#find-a-slot
 
-# Examples
+## Examples
 
-## Example 1: How imperative slotting API works in slotting=manual.
+### Example 1: How imperative slotting API works in slotAssignment=manual.
 
 ``` text
 host
-├──/shadowroot (slotting=manual)
+├──/shadowroot (slotAssignment=manual)
 │   ├── slot1
 │   └── slot2
 ├── A
 └── B
-
 ```
 
 ``` javascript
```

Comment:
The "== means ArrayEquals" is very confusing here since it means is isn't JavaScript code. It would be good to replace this all with `assert_array_equals`.

Reply:
Updated.